### PR TITLE
Allow user to limit the used parallelism for GHR

### PR DESCRIPTION
**Pull request by vchuravy**

### CODE REVIEW

**Comment by vchuravy** [comment]
Works in production xD, but didn't fix the issue.

File: src/AutoBuild.jl [hunk 1]
```diff
@@ -395,11 +395,11 @@ function get_compilers_versions(; compilers = [:c])
 end
 
 function upload_to_github_releases(repo, tag, path; gh_auth=Wizard.github_auth(;allow_anonymous=false),
-                                   attempts::Int = 3, verbose::Bool = false)
+                                   attempts::Int = 3, verbose::Bool = false, ncpu=Sys.CPU_THREADS)
```
Comment (staticfloat) [inline src/AutoBuild.jl:398]:
```suggestion
                                   attempts::Int = 3, verbose::Bool = false, num_parallel_uploads=Sys.CPU_THREADS)
```

File: src/AutoBuild.jl [hunk 2]
```diff
@@ -395,11 +395,11 @@ function get_compilers_versions(; compilers = [:c])
...
     for attempt in 1:attempts
         try
             ghr() do ghr_path
-                run(`$ghr_path -u $(dirname(repo)) -r $(basename(repo)) -t $(gh_auth.token) $(tag) $(path)`)
+                run(`$ghr_path -u $(dirname(repo)) -r $(basename(repo)) -t $(gh_auth.token) -p $(ncpu) $(tag) $(path)`)
```
Comment (staticfloat) [inline src/AutoBuild.jl:402]:
```suggestion
                run(`$ghr_path -u $(dirname(repo)) -r $(basename(repo)) -t $(gh_auth.token) -p $(num_parallel_uploads) $(tag) $(path)`)
```

**Review by staticfloat — APPROVED** [review]

**Comment by giordano** [comment]
What's the default parallelism in GHR?  The number of threads?

**Comment by vchuravy** [comment]
yeah I think so.

**Comment by vchuravy** [comment]
But we can also drop this PR.

**Comment by giordano** [comment]
> But we can also drop this PR.

Why?  It's ok with me if it resolves real problems, I just wanted to understand how this is changing in practice :slightly_smiling_face:

**Comment by vchuravy** [comment]
It was me losing my mind over unrelated errors and I was thinking that the rate limiting I was seeing was due to much parallelism.

**Comment by giordano** [comment]
Ok, let's close this then 🙂

**Pull request closed without merging**